romidata/webcache.py

```diff
@@ -0,0 +1,172 @@
+# -*- python -*-
+# -*- coding: utf-8 -*-
+# 
+# romidata - Data handling tools for the ROMI project
+# 
+# Copyright (C) 2018-2019 Sony Computer Science Laboratories
+# Authors: D. Colliaux, T. Wintz, P. Hanappe
+# 
+# This file is part of romidata.
+# 
+# romidata is free software: you can redistribute it
+# and/or modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation, either
+# version 3 of the License, or (at your option) any later version.
+# 
+# romidata is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU Lesser General Public
+# License along with romidata.  If not, see <https://www.gnu.org/licenses/>.
+# ------------------------------------------------------------------------------
+import json
+import os
+import hashlib
+from romidata import io
+from romidata import FSDB as DB
+from skimage.transform import resize
+import imageio
+import numpy as np
+
+def file_path(db, scanid, filesetid, fileid):
+    scan = db.get_scan(scanid)
+    fs = scan.get_fileset(filesetid)
+    f = fs.get_file(fileid)
+    return os.path.join(db.basedir, scan.id, fs.id, f.filename)
+
+def _hash(type, scanid, filesetid, fileid, size):
+    m = hashlib.sha1()
+    key = "%s|%s|%s|%s|%s" % (type, scanid, filesetid, fileid, size)
+    m.update(key.encode('utf-8'))
+    return m.hexdigest()
+
+# Image
+
+def image_hash(scanid, filesetid, fileid, size):
+    return _hash("image", scanid, filesetid, fileid, size)
+
+def image_resize(img, max_size):
+    i = np.argmax(img.shape[0:2])
+    if img.shape[i] <= max_size:
+        return img
+    if i == 0:
+        new_shape = [max_size, int(max_size * img.shape[1]/img.shape[0])]
+    else:
+        new_shape = [int(max_size * img.shape[0]/img.shape[1]), max_size]
+    return resize(img, new_shape)
+
+
+def image_cache(db, scanid, filesetid, fileid, size):
+    src = file_path(db, scanid, filesetid, fileid)
+    dir = os.path.join(db.basedir, scanid, "webcache")
+    os.makedirs(dir, exist_ok=True)
+    dst = os.path.join(dir, image_hash(scanid, filesetid, fileid, size))
+    
+    resolutions = {
+        "large": 1500,
+        "thumb": 150
+    }
+    maxsize = resolutions.get(size) 
+
+    image = imageio.imread(src)
+    # remove alpha channel
+    if image.shape[2] == 4:
+        image = image[:,:,:3]
```

Comment:
Would it be safer to use the `convert` method from `Pillow` to remove the alpha channel? 
https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.convert
Convert "RGBA" to "RGB" with:
```python
im.convert("RGB")
```